source/branching.rst

```diff
@@ -0,0 +1,135 @@
+to branch from it.  However, all feature branches must eventually merge back into ``develop`` or deleted.
+
+If the feature branch addresses a single issue, and is the only branch to address that issue, you
+can name the branch with the issue number as a mental cue. For example:
+
+    git checkout -b 42-my-new-feature develop
+
+Now that your branch is created, develop against it. Make as many commits as you'd like, and even
+as many pushes to origin as you'd like.  For example:
+
+    git add .
+    git commit -m "refactored models to address issue #42"
+    git push origin 42-my-new-feature -u
+
+Commit messages should be descriptive. Where practical, note changed files, why you changed them, and
+other metadata (WIP, not yet tested, addresses issue #42, etc.)
+
+
+Pull Requests
+-------------
+
+When you are certain of the following:
+
+- The branch has addressed the issue
+- All tests pass
+- Test coverage is above 80% for the entire repository
+- All code is PEP8 compliant
+- All migrations created if needed
+- All print/debug statements removed
+- The server runs as expected
+
+then you can do a final push and create a pull request.  GitHub makes this simple. There is a
+green button that reads "Review and create pull request."  You should look over the diff of the code
+and make sure it reads as you intend.  If so, add a comment to the pull request that it addresses issue #42.
+
+It is then incumbent on another member of the team to do the following:
+
+- Pull your feature branch
+- Run tests (this may be handled by a CI server such as TravisCI)
+- Run migrations (this may be handled by a CI server such as TravisCI)
+- Run their local dev server
+- Verify that the feature was addressed
+
+It is also incumbent upon the pull request reviewer to read the code diff and make comments on the code.
+Github presents a blue and white plus icon for adding line comments.
+
+When the reviewer agrees that the pull request is adequate, they should add a comment to the pull request.
+For fun, we typically add random emoticons such as ``:frog:``.
+
+By convention, the developer of the branch then merges it back into ``develop``, but other developers can do this step:
+
+    git checkout develop
+    git merge --no-ff 42-my-new-feature
+
+Note that GitHub allows you to merge and delete a branch from within the Pull Request GUI.
+
+
+Branch Lifespans
+-----------------
+
+Recall that ``origin/develop`` and ``origin/master`` are the only long term branches. Every other
+branch should be as short-lived as possible while still accomplishing its purpose. Dead branches
+(exploratory, or failed attempts to address an issue), or feature branches that drag on a long time,
+cloud the picture for other developers. Since this feature branch is no longer needed, it can be deleted
+via the "delete this branch" button on GitHub, or via the command line:
+
+    git branch -d 42-my-new-feature
+
```

Comment:
I've noted that the use of `--no-ff` in the merge can be used to preserve the original branch information in the commit.